File: output/CodeGen/avatar/Java/Python/atcoder_AGC006_B.py
def main(n):
    x = int(input())
    print(x, n)
    if x == 1 or x == n * 2 - 1:
        print("No")
        return
    print("Yes")
    sep = '\n'
    ans = IntStream.range(x + n - 1, x + n + n * 2 - 1).map(e -> (e % n + 1) + sep).collect(StringBuilder(), StringBuilder().append, StringBuilder().append).toString()
    print(ans)

File: output/ChatGPT/avatar/Java/C++/atcoder_AGC006_B.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

void print(int x, int n) {
    int max = n * 2 - 1;
    if (x == 1 || x == max) {
        cout << "No" << endl;
        return;
    }
    cout << "Yes" << endl;
    string ans;
    for (int i = x + n - 1; i < x + n + max - 1; i++) {
        ans += to_string((i % max) + 1);
        ans += "\n";
    }
    cout << ans;
}

int main() {
    int n, x;
    cin >> n >> x;
    print(x, n);
    return 0;
}